### PR TITLE
[SCB-2533] [1.3.x] encode invocation context before add to http header

### PR DESCRIPTION
**Pull request by demonbug**
fix issus #3008

add writeUnicodeValueAsString to JsonUtils to support unicode encoding
encode invocation context before add to http header

### CODE REVIEW

File: transports/transport-rest/transport-rest-client/src/main/java/org/apache/servicecomb/transport/rest/client/http/RestClientInvocation.java [hunk 1]
```diff
@@ -274,7 +274,7 @@ protected void fail(ConnectionBase connection, Throwable e) {
 
   protected void setCseContext() {
     try {
-      String cseContext = JsonUtils.writeValueAsString(invocation.getContext());
+      String cseContext = JsonUtils.writeUnicodeValueAsString(invocation.getContext());
```
Comment (liubao68) [inline transports/transport-rest/transport-rest-client/src/main/java/org/apache/servicecomb/transport/rest/client/http/RestClientInvocation.java:277]: Is this compatible with old versions? e.g. new consumer invoke an older version provider.

Comment (demonbug) [reply transports/transport-rest/transport-rest-client/src/main/java/org/apache/servicecomb/transport/rest/client/http/RestClientInvocation.java:277]: I think it is compatible, Jackson will automatic decode unicode string in deserialization.
In testcase, use writeUnicodeValueAsString to serialize and use the old OBJ_MAPPER to deserialize, the value is not changed

Comment (liubao68) [reply transports/transport-rest/transport-rest-client/src/main/java/org/apache/servicecomb/transport/rest/client/http/RestClientInvocation.java:277]: Can you add some unit test case for this?

File: foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/utils/JsonUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import junit.framework.TestCase;
+import org.apache.commons.lang.StringUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
```
Comment (liubao68) [inline foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/utils/JsonUtilsTest.java:28]: Use junit5 for test cases, and I'd prefer PR for master.

Comment (demonbug) [reply foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/utils/JsonUtilsTest.java:28]: the lastest NUWA is still using branch 1.3.x, and this branch has not import junit5

File: transports/transport-rest/transport-rest-client/src/main/java/org/apache/servicecomb/transport/rest/client/http/RestClientInvocation.java (hunk 1 above)
Comment (demonbug) [reply transports/transport-rest/transport-rest-client/src/main/java/org/apache/servicecomb/transport/rest/client/http/RestClientInvocation.java:277]: ok, i added 2 testcase in TestRestClientInvocation.java

**Review by Shoothzj — CHANGES_REQUESTED** [review]
Apache commons-lang3 also have `StringEscapeUtils`, why we need `commons-text`?

**Comment by demonbug** [comment]
> Apache commons-lang3 also have `StringEscapeUtils`, why we need `commons-text`?
StringEscapeUtils was deprecated in commons-lang3

**Review by Shoothzj — APPROVED** [review]

**Review by liubao68 — APPROVED** [review]

**Pull request merged**